Source/santasyncservice/SNTSyncStage.mm

```diff
@@ -202,7 +202,9 @@ - (NSData *)dataFromRequest:(NSURLRequest *)request
       errStr = [NSHTTPURLResponse localizedStringForStatusCode:response.statusCode];
     } else {
       code = (long)requestError.code;
+      if (code == 0) code = response.StatusCode;
       errStr = requestError.localizedDescription;
+      if (!errStr) errStr = @"Unknown Error";
     }
     LOGE(@"HTTP Response: %ld %@", code, errStr);
     if (error != NULL) {
```

Comment:
As a backstop we should also check `errStr` is not nil before creating `@{NSLocalizedDescriptionKey : errStr}`.